src/main/java/hudson/remoting/PingThread.java

```diff
@@ -61,11 +63,14 @@
      */
     private final long interval;
 
-    public PingThread(Channel channel, long timeout, long interval) {
+    private final int maxTimeouts;
+
+    public PingThread(Channel channel, long timeout, long interval, int maxTimeouts) {
```

Comment:
Instead of changing a public constructor, create a new one and set your default within the old one. I fear other plugins might use the current constructor you are replacing.

Reply:
Looks like this has been resolved.